src/Symfony/Component/Form/Form.php

```diff
@@ -562,6 +562,7 @@ public function submit($submittedData, $clearMissing = true)
 
                 foreach ($this->children as $name => $child) {
                     $isSubmitted = array_key_exists($name, $submittedData);
+                    $clearMissing = $this->getConfig()->getOption('expanded') ?: $clearMissing;
```

Comment:
@fabpot unless I missed something else, `$clearMissing` in only used as an argument of `FormType::submit()` and there is no way no access its behaviour from a type definition, or it would need to override this method.

Handling a custom submit process for a particular form type would require a big refactoring...

@webmozart any thoughts ?